src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -468,6 +449,37 @@ private void PublishDiagnosticsNotification(string uri, string expression, Expre
                 }));
         }
 
+        /// <summary>
+        /// Construct a Range based on a Span for a given expression.
+        /// </summary>
+        /// <param name="expression">The expression.</param>
+        /// <param name="span">The Span.</param>
+        /// <returns>Generated Range.</returns>
+        public static Range GetRange(string expression, Span span)
```

Comment:
>GetRange [](http://example.com/codeflow?start=28&length=8)

Can we add Contract.Asserts here?

(startLine < endLine) || 
(startLine == endLine  && startChar <= endChar)